flutter_local_notifications/test/flutter_local_notifications_test.dart

```diff
@@ -14,6 +16,44 @@ void main() {
     expect(const DarwinNotificationAttachment(''),
         isA<DarwinNotificationAttachment>());
   });
+
+  group('providesAppNotificationSettings tests', () {
+    test('initialization defaults to false', () {
+      final DarwinInitializationSettings settings =
+          DarwinInitializationSettings();
+      expect(settings.requestProvidesAppNotificationSettings, false);
+    });
+
+    test('initialization can be set to true', () {
+      final DarwinInitializationSettings settings =
+          DarwinInitializationSettings(
+        requestProvidesAppNotificationSettings: true,
+      );
+      expect(settings.requestProvidesAppNotificationSettings, true);
+    });
+
+    test('maps correctly to platform specific map', () {
+      final DarwinInitializationSettings settings =
+          DarwinInitializationSettings(
+        requestProvidesAppNotificationSettings: true,
+      );
+      final Map<String, Object> map = settings.toMap();
+      expect(map['requestProvidesAppNotificationSettings'], true);
+    });
+
+    test('permission check returns correct value', () async {
+      final NotificationsEnabledOptions options = NotificationsEnabledOptions(
+        isEnabled: true,
+        isSoundEnabled: true,
+        isAlertEnabled: true,
+        isBadgeEnabled: true,
+        isProvisionalEnabled: false,
+        isCriticalEnabled: false,
+        isProvidesAppNotificationSettingsEnabled: true,
+      );
+      expect(options.isProvidesAppNotificationSettingsEnabled, true);
```

Comment:
it says it should be executing a permissions check and the test should be asserting what it returns but the test doesn't do this. is this test missing some logic?